diff -ur rpm-4.16.1.2/fileattrs/python.attr rpm-4.16.1.2.x32/fileattrs/python.attr
--- rpm-4.16.1.2/fileattrs/python.attr	2020-08-31 11:14:07.991087349 +0200
+++ rpm-4.16.1.2.x32/fileattrs/python.attr	2021-03-21 21:50:04.127398387 +0100
@@ -14,14 +14,15 @@
 %__python_requires() %{lua:
     -- Match buildroot paths of the form
     --    /PATH/OF/BUILDROOT/usr/lib/pythonMAJOR.MINOR/  and
-    --    /PATH/OF/BUILDROOT/usr/lib64/pythonMAJOR.MINOR/
+    --    /PATH/OF/BUILDROOT/usr/lib64/pythonMAJOR.MINOR/ and
+    --    /PATH/OF/BUILDROOT/usr/libx32/pythonMAJOR.MINOR/
     -- generating a line of the form:
     --    python(abi) = MAJOR.MINOR
     local path = rpm.expand('%1')
-    if path:match('/usr/lib%d*/python%d+%.%d+/.*') then
-        local requires = path:gsub('.*/usr/lib%d*/python(%d+%.%d+)/.*', 'python(abi) = %1')
+    if path:match('/usr/lib%w*/python%d+%.%d+/.*') then
+        local requires = path:gsub('.*/usr/lib%w*/python(%d+%.%d+)/.*', 'python(abi) = %1')
         print(requires)
     end
 }
 
-%__python_path ^((%{_prefix}/lib(64)?/python[[:digit:]]+\\.[[:digit:]]+/.*\\.(py[oc]?|so))|(%{_bindir}/python[[:digit:]]+\\.[[:digit:]]+))$
+%__python_path ^((%{_prefix}/lib(64|x32)?/python[[:digit:]]+\\.[[:digit:]]+/.*\\.(py[oc]?|so))|(%{_bindir}/python[[:digit:]]+\\.[[:digit:]]+))$
diff -ur rpm-4.16.1.2/fileattrs/pythondist.attr rpm-4.16.1.2.x32/fileattrs/pythondist.attr
--- rpm-4.16.1.2/fileattrs/pythondist.attr	2020-05-28 12:04:25.026136626 +0200
+++ rpm-4.16.1.2.x32/fileattrs/pythondist.attr	2021-03-21 21:50:04.127398387 +0100
@@ -1,3 +1,3 @@
 %__pythondist_provides	%{_rpmconfigdir}/pythondistdeps.py --provides --majorver-provides
 %__pythondist_requires	%{_rpmconfigdir}/pythondistdeps.py --requires
-%__pythondist_path		/lib(64)?/python[[:digit:]]\\.[[:digit:]]+/site-packages/[^/]+\\.(dist-info|egg-info|egg-link)$
+%__pythondist_path		/lib(64|x32)?/python[[:digit:]]\\.[[:digit:]]+/site-packages/[^/]+\\.(dist-info|egg-info|egg-link)$
